### PR TITLE
added ability to accept and decline reservations

### PR DESCRIPTION
**Pull request by eninodel**

### CODE REVIEW

File: GearRent/ViewControllers/ListingReservationsViewController.m [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//
+//  ListingReservationsViewController.m
+//  GearRent
+//
+//  Created by Edwin Delgado on 7/14/22.
+//
+
+#import "ListingReservationsViewController.h"
+#import "../Models/Item.h"
```
Comment (alonbee) [inline GearRent/ViewControllers/ListingReservationsViewController.m:9]: i guess you can just use "Item.h", applied to other header import

File: GearRent/ViewControllers/ListingReservationsViewController.m [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            NSLog(@"END: Error fetching reservation dates");
+        }
+    }];
+}
+
+- (IBAction)didTapBack:(id)sender {
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
```
Comment (alonbee) [inline GearRent/ViewControllers/ListingReservationsViewController.m:47]: no need for the `nonnull` in the the implementations

Comment (eninodel) [reply GearRent/ViewControllers/ListingReservationsViewController.m:47]: This is the default method header xcode gave me?

File: GearRent/ViewControllers/DetailsViewController.m [hunk 3]
```diff
@@ -164,17 +167,18 @@ - (void)calendar:(JTCalendarManager *)calendar prepareDayView:(JTCalendarDayView
     }
 }
 
-- (Boolean) isDateReserved: (NSDate *) date{
+- (Boolean)isDateReserved:(NSDate *) date{
     for(int i = 0; i < self.datesReserved.count; i++){
         NSDateInterval *interval = self.datesReserved[i];
-        if([interval containsDate:date] == YES){
+        Reservation *reservation = (Reservation *) self.reservations[i];
+        if([interval containsDate:date] == YES && [reservation.status isEqualToString:@"ACCEPTED"]){
```
[the remark this answers is not in the document]
Comment (eninodel) [reply GearRent/ViewControllers/DetailsViewController.m:174]: Will do in refactor, clean up later like mentioned earlier in meeting

File: GearRent/ViewControllers/ReservationTableViewCell.m [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+//  ReservationTableViewCell.m
+//  GearRent
+//
+//  Created by Edwin Delgado on 7/14/22.
+//
+
+#import "ReservationTableViewCell.h"
+
+@interface ReservationTableViewCell()
+@property (weak, nonatomic) IBOutlet UILabel *leaseeLabel;
```
Comment (alonbee) [inline GearRent/ViewControllers/ReservationTableViewCell.m:11]: new line

File: GearRent/ViewControllers/ReservationTableViewCell.m [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+@property (weak, nonatomic) IBOutlet UILabel *datesLeasedLabel;
+@property (weak, nonatomic) IBOutlet UILabel *currentStatusLabel;
+@property (weak, nonatomic) IBOutlet UIButton *declineButton;
+@property (weak, nonatomic) IBOutlet UIButton *acceptButton;
+
+- (IBAction)didAccept:(id)sender;
+- (IBAction)didDecline:(id)sender;
+@end
```
Comment (alonbee) [inline GearRent/ViewControllers/ReservationTableViewCell.m:19]: new line

File: GearRent/ViewControllers/ReservationTableViewCell.m [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+@implementation ReservationTableViewCell
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    // Initialization code
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
```
Comment (alonbee) [inline GearRent/ViewControllers/ReservationTableViewCell.m:28]: we probably don't need this if it's just calling the super

File: GearRent/ViewControllers/ReservationTableViewCell.m [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    self.currentStatusLabel.text = self.reservation.status;
+    NSString *datesLeasedString = @"Dates leased: ";
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.dateFormat = @"dd-MM-yyyy";
+    NSLog(@"%@", self.reservation.dates.startDate);
+    datesLeasedString = [datesLeasedString stringByAppendingString: [dateFormatter stringFromDate:self.reservation.dates.startDate]];
+    datesLeasedString = [datesLeasedString stringByAppendingString:@" - "];
+    datesLeasedString = [datesLeasedString stringByAppendingString: [dateFormatter stringFromDate:self.reservation.dates.endDate]];
+    self.datesLeasedLabel.text = datesLeasedString;
+    PFQuery *query = [PFUser query];
```
Comment (alonbee) [inline GearRent/ViewControllers/ReservationTableViewCell.m:44]: might be good to setup a new folder for API related files & functions.

**Pull request merged**